presto-docs/src/main/sphinx/presto_cpp/properties.rst

```diff
@@ -0,0 +1,140 @@
+    optimizer.optimize-hash-generation=false
+    regex-library=RE2J
+    use-alternative-function-signatures=true
+    experimental.table-writer-merge-operator-enabled=false
+
+``experimental.table-writer-merge-operator-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``true``
+
+  Merge TableWriter output before sending to TableFinishOperator.
+
+``native-execution-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``false``
+
+  This property is required when running Presto C++ workers because of 
+  underlying differences in behavior from Java workers.
+
+``optimizer.optimize-hash-generation``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``true``
+
+  Set this property to ``false`` when running Presto C++ workers 
+  because Velox does not support optimized hash generation, 
+  instead performing this optimization adaptively.
+
+``regex-library``
+^^^^^^^^^^^^^^^^^
+
+* **Type:** ``type``
+* **Allowed values:** ``RE2J``
+* **Default value:** ``JONI``
+
+  Only `RE2J <https://github.com/google/re2j>`_ is currently supported by Velox.
+
+``use-alternative-function-signatures``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``false``
+
+  Some aggregation functions use generic intermediate types which are 
+  not compatible with Velox aggregation function intermediate types. One  
+  example function is ``approx_distinct``, whose intermediate type is 
+  ``VARBINARY``. 
+  Stubs in the Presto codebase provide alternative stub implementations 
+  which return the correct intermediate type.
+
+Worker Properties
+-----------------
+
+The configuration properties of Presto C++ workers. 
+
+``async-data-cache-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``true``
+
+  In-memory cache.
+
+``query.max-memory-per-node``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``integer``
+* **Default value:** ``4GB``
+
+  Max memory usage for each query.
+
+``query-memory-gb``
+^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``integer``
+* **Default value:** ``38``
+
+  The total memory capacity that can be used across all query executions. 
+  Memory for system usage such as disk spilling and cache prefetch which 
+  are not counted in query memory usage.
+
+``query-reserved-memory-gb``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``integer``
+* **Allowed values:** ``0``
+* **Default value:** ``4``
+
+  Value must be set to ``0``. Specifies the amount of query memory capacity reserved 
+  to ensure that each query has the minimal memory capacity to run. A query can 
+  only allocate from the reserved query memory if its current capacity is less 
+  than the minimal memory capacity as specified by 
+  ``memory-pool-reserved-capacity``. 
+  
+  The exceeding capacity must allocate from the non-reserved query memory.
+
+``system-memory-gb``
```

Comment:
This is very important to be configured right. New users often overlook this. They must set this to the available machine memory.
I see that the configs are ordered alphabetically. Can we add a note indicating that users must set this according to their deployment?

Reply:
I will see if we can automate the inference of this value similar to `task.max-drivers-per-task` below.

Reply:
> This is very important to be configured right. New users often overlook this. They must set this to the available machine memory. I see that the configs are ordered alphabetically. Can we add a note indicating that users must set this according to their deployment?

Of course we can add text to address this.

Reply:
> I will see if we can automate the inference of this value similar to `task.max-drivers-per-task` below.

@majetideepak if you do automate the inference of this value, let me know if this property entry should then be removed from the documentation.